dataactvalidator/validation_handlers/file_generation_manager.py

```diff
@@ -27,129 +20,92 @@ class FileGenerationManager:
-            if self.is_local:
-                self.job.filename = "".join([CONFIG_BROKER['broker_files'], self.job.original_filename])
-            else:
-                self.job.filename = "".join([str(self.job.submission_id), "/", self.job.original_filename])
-            self.sess.commit()
-
-            # Generate the file, and upload to S3
-            if self.job.file_type.letter_name in ['D1', 'D2']:
-                # Update the validation Job if necessary
-                update_validation_job_info(self.sess, self.job)
-
-                self.generate_d_file()
-            elif self.job.file_type.letter_name == 'E':
-                self.generate_e_file()
-            else:
-                self.generate_f_file()
-
-            mark_job_status(self.job.job_id, 'finished')
-
-        logger.info({
-            'message': 'Finished file {} generation'.format(self.job.file_type.letter_name),
-            'message_type': 'ValidatorInfo', 'job_id': self.job.job_id, 'agency_code': self.agency_code,
-            'file_type': self.job.file_type.letter_name, 'start_date': self.job.start_date,
-            'end_date': self.job.end_date, 'filename': self.job.original_filename
-        })
-
-    def generate_d_file(self):
-        """ Write file D1 or D2 to an appropriate CSV. """
-        log_data = {'message': 'Starting file {} generation'.format(self.job.file_type.letter_name),
-                    'message_type': 'ValidatorInfo', 'job_id': self.job.job_id, 'agency_code': self.agency_code,
-                    'file_type': self.job.file_type.letter_name, 'start_date': self.job.start_date,
-                    'end_date': self.job.end_date, 'filename': self.job.original_filename}
-        if self.job.submission_id:
-            log_data['submission_id'] = self.job.submission_id
+        self.file_generation = file_generation
+        self.job = job
+        self.file_type = job.file_type.letter_name if job else file_generation.file_type
+
+    def generate_file(self):
+        """ Generates a file based on the FileGeneration object and updates any Jobs referencing it """
+        raw_filename = CONFIG_BROKER["".join([FILE_TYPE_DICT_LETTER_NAME[self.file_type], "_file_name"])]
+        file_name = S3Handler.get_timestamped_filename(raw_filename)
+        if self.is_local:
+            file_path = "".join([CONFIG_BROKER['broker_files'], file_name])
+        else:
+            file_path = "".join(["None/", file_name])
+
+        # Generate the file and upload to S3
+        if self.file_generation:
+            self.generate_d_file(file_path)
+        elif self.file_type == 'E':
+            self.generate_e_file()
+        else:
+            self.generate_f_file()
+
+        log_data = {
+            'message': 'Finished file {} generation'.format(self.file_type), 'message_type': 'ValidatorInfo',
+            'file_type': self.file_type, 'file_path': file_path
+        }
+        if self.file_generation:
```

Comment:
Might be more efficient and safe to not have to exercise this condition twice, and put the specialized log data inside the `if` the first time this was checked above, plus declaring `log_data` with defaults before the above condition checks.